lib/components/map/connected-stop-viewer-overlay.js

```diff
@@ -1,139 +1,104 @@
-        <ParkAndRideOverlay
-          parkAndRideLocations={parkAndRideLocations}
-          setLocation={setLocation}
-          visible
-        />
-        {mapConfig.overlays &&
-          mapConfig.overlays.map((overlayConfig, k) => {
-            switch (overlayConfig.type) {
-              case 'bike-rental':
-              case 'otp2-bike-rental':
-                return (
-                  <VehicleRentalOverlay
-                    key={k}
-                    {...overlayConfig}
-                    configCompanies={configCompanies}
-                    stations={bikeRental && bikeRental.stations}
-                    visible={zoom >= MIN_ZOOM - 1}
-                  />
-                )
-              case 'micromobility-rental':
-              case 'otp2-micromobility-rental':
-                return (
-                  <VehicleRentalOverlay
-                    key={k}
-                    {...overlayConfig}
-                    configCompanies={configCompanies}
-                    stations={vehicleRental?.stations}
-                    visible={zoom >= MIN_ZOOM - 1}
-                  />
-                )
-              default:
-                return null
-            }
-          })}
-      </FeatureGroup>
-    )
-  }
+        ))}
+      <ParkAndRideOverlay
+        parkAndRideLocations={parkAndRideLocations}
+        setLocation={setLocation}
+        visible
+      />
+      {mapConfig.overlays &&
+        mapConfig.overlays.map((overlayConfig, k) => {
+          switch (overlayConfig.type) {
+            case 'bike-rental':
+            case 'otp2-bike-rental':
+              return (
+                <VehicleRentalOverlay
+                  {...overlayConfig}
+                  configCompanies={configCompanies}
+                  key={k}
+                  stations={bikeRental?.stations}
+                />
+              )
+            case 'micromobility-rental':
+            case 'otp2-micromobility-rental':
+              return (
+                <VehicleRentalOverlay
+                  {...overlayConfig}
+                  configCompanies={configCompanies}
+                  key={k}
+                  stations={vehicleRental?.stations}
+                />
+              )
+            default:
+              return null
+          }
+        })}
+    </>
+  )
 }
 
 // connect to the redux store
 
 const mapStateToProps = (state) => {
   const stopLookup = state.otp.transitIndex.stops
   const stopData = stopLookup[state.otp.ui.viewedStop?.stopId]
-  const nearbyStops = stopData?.nearbyStops?.map((stopId) => stopLookup[stopId])
-  const stops = []
-  if (stopData) stops.push(stopData)
-  if (nearbyStops && nearbyStops.length > 0) stops.push(...nearbyStops)
+  const nearbyStops =
+    stopData?.nearbyStops?.map((stopId) => stopLookup[stopId]) || []
   return {
     configCompanies: state.otp.config.companies,
     mapConfig: state.otp.config.map,
     stopData,
-    stops
+    stops: [...nearbyStops, stopData]
```

Comment:
This is a little confusing... is stopData an array or an object?

Reply:
stopData is a single stop, nearbyStops is a list of stops. We're not doing `flatMap`, so I think this is clear right? Is there something we can do to make this clearer?

Reply:
I was not understanding that stopData is a single stop!